Allow fixed width integer types as bit fields

tadeuzagallo opened the pull request:
#### b831bf8d6468696e8854fc8da2c24d426342eb86
<pre>
Allow fixed width integer types as bit fields
<a href="https://bugs.webkit.org/show_bug.cgi?id=252475">https://bugs.webkit.org/show_bug.cgi?id=252475</a>
rdar://105593556

Reviewed by David Kilzer.

Currently, the style checker complains about uint8_t and friends as bit fields,
claiming that it&apos;s not an unsigned integral type. This patch updates the rule
to accept all intN_t and uintN_t types.

* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_language):
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(CppStyleTest):

Canonical link: <a href="https://commits.webkit.org/260878@main">https://commits.webkit.org/260878@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d2d81687f07b84c80d9ad332cb4f77575fa4b5e7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109766 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18881 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42503 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1236 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118885 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/113717 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/20352 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10078 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102034 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115514 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/15152 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/98383 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43374 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/113223 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/97120 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30036 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/85169 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11607 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31378 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12259 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/8317 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17628 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50987 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7542 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14009 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

ddkilzer commented on Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
```diff
@@ -3206,6 +3206,16 @@ def test_enum_bitfields(self):
+        self.assert_lint('uint8_t a : 1;', '')
+        self.assert_lint('uint16_t a : 1;', '')
+        self.assert_lint('uint32_t a : 1;', '')
+        self.assert_lint('uint32_t a : 1;', '')
```
Duplicate line?

ddkilzer commented on Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
```diff
@@ -3206,6 +3206,16 @@ def test_enum_bitfields(self):
+        self.assert_lint('int8_t a : 1;', '')
+        self.assert_lint('int16_t a : 1;', '')
+        self.assert_lint('int32_t a : 1;', '')
+        self.assert_lint('int32_t a : 1;', '')
```
Duplicate line?

ddkilzer commented on Tools/Scripts/webkitpy/style/checkers/cpp.py:
```diff
@@ -4022,7 +4022,7 @@ def check_language(filename, clean_lines, line_number, file_extension, include_s
     # Check for plain bitfields declared without either "singed" or "unsigned".
```
Nit:  Could fix typo "singed" while we're here.

ddkilzer reviewed (approved):
r=me, but please fix the duplicate test lines before landing.  Thanks!

tadeuzagallo committed:
Allow fixed width integer types as bit fields
https://bugs.webkit.org/show_bug.cgi?id=252475
rdar://105593556

Reviewed by David Kilzer.

Currently, the style checker complains about uint8_t and friends as bit fields,
claiming that it's not an unsigned integral type. This patch updates the rule
to accept all intN_t and uintN_t types.

* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_language):
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(CppStyleTest):

Canonical link: https://commits.webkit.org/260878@main
